services/common/rs_server_common/data_retrieval/eodag_provider.py

```diff
@@ -0,0 +1,80 @@
+"""EODAG Provider."""
+from pathlib import Path
+
+from eodag import EODataAccessGateway, EOProduct
+
+from .provider import CreateProviderFailed, Product, Provider, TimeRange
+
+
+class EodagProvider(Provider):
+    """An EODAG provider.
+
+    It uses EODAG to provide data from external sources.
+    """
+
+    def __init__(self, config_file: str, provider: str):
```

Comment:
Why do you switch the config_file type from Path to str ?
Manipulating Path for paths let us use some helper functions
and avoid to have to handle a lot of degraded cases.

Reply:
If you keep this change, you should update the docstring
and the unit tests.

Reply:
Fixed, config_file can be a PosixPath.